### PR TITLE
Fix hrefLang + small revamp of i18n data

### LINKED ISSUE #302: TypeError: Cannot read property 'locale' of undefined
## Description

When spinning up a site without the `i18n.additionalLocales` option, an error occurs:

```
 ERROR #11321  PLUGIN

"@newrelic/gatsby-theme-newrelic" threw an error while running the sourceNodes lifecycle:

Cannot read property 'locale' of undefined

  130 |
  131 |   i18n.locales.forEach((locale) => {
> 132 |     const isDefault = locale.locale === defaultLocale.locale;
      |                              ^
  133 |
  134 |     const data = {
  135 |       ...locale,

File: node_modules/@newrelic/gatsby-theme-newrelic/gatsby-node.js:132:30

TypeError: Cannot read property 'locale' of undefined

  - gatsby-node.js:132
    [developer-website]/[@newrelic]/gatsby-theme-newrelic/gatsby-node.js:132:30

  - Array.forEach

  - gatsby-node.js:131 Object.exports.sourceNodes
    [developer-website]/[@newrelic]/gatsby-theme-newrelic/gatsby-node.js:131:16

  - api-runner-node.js:469 runAPI
    [developer-website]/[gatsby]/src/utils/api-runner-node.js:469:22

  - api-runner-node.js:620 Promise.catch.decorateEvent.pluginName
    [developer-website]/[gatsby]/src/utils/api-runner-node.js:620:13

  - debuggability.js:384 Promise._execute
    [developer-website]/[bluebird]/js/release/debuggability.js:384:9

  - promise.js:518 Promise._resolveFromExecutor
    [developer-website]/[bluebird]/js/release/promise.js:518:18

  - promise.js:103 new Promise
    [developer-website]/[bluebird]/js/release/promise.js:103:10

  - api-runner-node.js:618
    [developer-website]/[gatsby]/src/utils/api-runner-node.js:618:16

  - util.js:16 tryCatcher
    [developer-website]/[bluebird]/js/release/util.js:16:23

  - reduce.js:166 Object.gotValue
    [developer-website]/[bluebird]/js/release/reduce.js:166:18

  - reduce.js:155 Object.gotAccum
    [developer-website]/[bluebird]/js/release/reduce.js:155:25

  - util.js:16 Object.tryCatcher
    [developer-website]/[bluebird]/js/release/util.js:16:23

  - promise.js:547 Promise._settlePromiseFromHandler
    [developer-website]/[bluebird]/js/release/promise.js:547:31

  - promise.js:604 Promise._settlePromise
    [developer-website]/[bluebird]/js/release/promise.js:604:18

  - promise.js:649 Promise._settlePromise0
    [developer-website]/[bluebird]/js/release/promise.js:649:10
```

## Steps to Reproduce

* Don't specify `i18n.additionalLocales` option in `gatsby-config.js`. 
* Run the site via `yarn start`
* See the error

## Expected Behaviour

We should not care if `additionalLocales` are not defined. This should default to an empty array.

### PR DESCRIPTION
**Pull request by jerelmiller**
Closes #302

## Description

Simplifies the configuration for i18n locales. The theme will now do the heavy
lifting of matching the locale with the locale data. Config will now use an
array of strings instead of a full configuration so that the theme can match the
appropriate data.

This PR also does some refactoring to use a getI18nConfig helper everywhere.
This should make it much easier to work with i18n config as the previous
withDefaults helper became pretty confusing.

### CODE REVIEW

File: packages/gatsby-theme-newrelic/src/components/SEO.js [hunk 1]
```diff
@@ -21,39 +21,40 @@ const SEO = ({ title, location, children }) => {
...
   return (
     <Helmet titleTemplate={template}>
+      <html lang={locale.hrefLang} />
       <title>{title || defaultTitle}</title>
-      {locales.map(({ locale, localizedPath }, i) => {
+      <link rel="canonical" href={new URL(location.pathname, siteUrl).href} />
+      {locales.map(({ hrefLang, isDefault, locale }) => {
+        const url = new URL(
+          path.join(isDefault ? '/' : locale, subPath),
+          'https://jerelmiller.ngrok.io'
```
Comment (jerelmiller) [inline packages/gatsby-theme-newrelic/src/components/SEO.js:49]: Oops!

Comment (caylahamann) [inline packages/gatsby-theme-newrelic/src/components/SEO.js]: I assume this was for testing :)

**Review by caylahamann — APPROVED** [review]
These all look good to me :) I really like the changes that were made, I feel like it simplifies things a lot

File: packages/gatsby-theme-newrelic/src/components/SEO.js (hunk 1 above)
Comment (jerelmiller) [reply packages/gatsby-theme-newrelic/src/components/SEO.js]: Haha yep! I caught it as well. its been fixed

**Review by zstix — APPROVED** [review]

**Pull request merged**